File: src/main/java/studio/karllang/karl/lib/Operators.java
package studio.karllang.karl.lib;

import studio.karllang.karl.lexer.TokenType;

import java.util.HashMap;

public class Operators {
    private static final HashMap<String, TokenType> operators = new HashMap<>();

    static {
        operators.put("+", TokenType.PLUS);
        operators.put("-", TokenType.MINUS);
        operators.put("*", TokenType.MULTIPLY);
        operators.put("/", TokenType.DIVIDE);
        operators.put("%", TokenType.MODULO);
    }

    public static boolean isOperator(TokenType type) {
        return operators.containsValue(type);
    }
}

File: src/main/kotlin/studio/karllang/karl/lib/Operators.kt
package studio.karllang.karl.lib

import studio.karllang.karl.lexer.TokenType

class Operators {
    companion object {
        private val operators = HashMap<String, TokenType>()

        init {
            operators["+"] = TokenType.PLUS
            operators["-"] = TokenType.MINUS
            operators["*"] = TokenType.MULTIPLY
            operators["/"] = TokenType.DIVIDE
            operators["%"] = TokenType.MODULO
        }

        @JvmStatic
        fun isOperator(type: TokenType): Boolean {
            return operators.containsValue(type)
        }
    }
}
